brainframe/api/bf_errors.py

```diff
@@ -290,3 +290,9 @@ class RemoteConnectionError(BaseAPIError):
     """The server encountered an error while connecting to a remote resource
     that is required for the requested operation.
     """
+
+
+@_register_error()
+class ServerNotReadyError(BaseAPIError):
```

Comment:
@_register_error() registers exceptions in a dict that maps them to their name. This is used to match error messages from the server to these exception classes. As far as I understand it, the server will never respond to the client with a ServerNotReadyError, right? This is something that the client creates. If so, this decorator is not necessary.

Reply:
Ooh, good catch. Maybe a better name for that function would be _register_api_error

Reply:
They're _all_ API errors, just some are REST API errors.

Maybe `@_server_origin_error`?

Reply:
Fine by me!

Reply:
So here's a dumb question... why not just retrieve `BaseAPIError.__subclasses__` instead of using a decorator?

Reply:
The decorator lets you specify a name other than the class name in cases where they don't match up. For example, NotImplementedError in the REST API becomes NotImplementedInAPIError to avoid clashing with Python's built-in exception with the same name. Also, it let us rename the errors with "Plugin" in the name to "Capsule" even though the server hasn't been updated with that change yet.

Reply:
Also, I didn't know `__subclasses__` existed :stuck_out_tongue:

Reply:
Changes made